File: android_api_java/src/main/java/com/cn/android/wifi/WifiApi.java
package com.cn.android.wifi;

import android.content.Context;
import android.net.wifi.ScanResult;
import android.net.wifi.WifiInfo;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

import kotlin.reflect.KProperty1;

public interface WifiApi {

    boolean enabledWifi(Context context);

    boolean enabledWifi(Context context, boolean enable);

    boolean isWifiEnabled(Context context);

    boolean startScan(Context context);

    List<ScanResult> getScanResults(Context context);

    boolean connectWifi(Context context, ScanResult result, String password);

    WifiInfo getConnectionWifiInfo(Context context);
}

File: android_api_kotlin/src/main/java/com/cn/android/wifi/WifiApi.kt
package com.cn.android.wifi

import android.content.Context
import android.net.wifi.ScanResult
import android.net.wifi.WifiInfo

interface WifiApi {

    fun enabledWifi(context: Context): Boolean

    fun enabledWifi(context: Context, boolean: Boolean): Boolean

    fun isWifiEnabled(context: Context): Boolean

    fun startScan(context: Context): Boolean

    fun getScanResults(context: Context): List<ScanResult>?

    fun connectWifi(context: Context, result: ScanResult, password: String): Boolean

    fun getCurrentWifiInfo(context: Context): WifiInfo?
}